### ISSUE TITLE
Error message Loop with MyPaint 1.3.0

### ISSUE THREAD

**Issue by DigitalKO** [mypaint/mypaint#751]
#### Description of the problem

I compiled, from git master, MyPaint 1.3.0-alpha+git.2c07a27b.

I also compile libmypaint and got these packages:
```
gir1.2-libmypaint0_1.3.0-0dev2_amd64.deb
libmypaint0_1.3.0-0dev2_amd64.deb
libmypaint0-dbgsym_1.3.0-0dev2_amd64.deb
libmypaint-dev_1.3.0-0dev2_amd64.deb
```

When I have installed these packages, I got this output (with a "little" error message):

```
dpkg -i  *.deb
(Reading database ... 414394 files and directories currently installed.)
Preparing to unpack gir1.2-libmypaint0_1.3.0-0dev2_amd64.deb ...
Unpacking gir1.2-libmypaint0 (1.3.0-0dev2) ...
dpkg: error processing archive gir1.2-libmypaint0_1.3.0-0dev2_amd64.deb (--install):
 trying to overwrite '/usr/lib/girepository-1.0/AccountsService-1.0.typelib', which is also in package gir1.2-accountsservice-1.0 0.6.40-4
dpkg-deb: error: subprocess paste was killed by signal (Broken pipe)
Preparing to unpack libmypaint0_1.3.0-0dev2_amd64.deb ...
Unpacking libmypaint0 (1.3.0-0dev2) ...
Selecting previously unselected package libmypaint0-dbgsym.
Preparing to unpack libmypaint0-dbgsym_1.3.0-0dev2_amd64.deb ...
Unpacking libmypaint0-dbgsym (1.3.0-0dev2) ...
Selecting previously unselected package libmypaint-dev.
Preparing to unpack libmypaint-dev_1.3.0-0dev2_amd64.deb ...
Unpacking libmypaint-dev (1.3.0-0dev2) ...
Setting up libmypaint0 (1.3.0-0dev2) ...
Setting up libmypaint0-dbgsym (1.3.0-0dev2) ...
Setting up libmypaint-dev (1.3.0-0dev2) ...
Processing triggers for libc-bin (2.24-8) ...
Errors were encountered while processing:
 gir1.2-libmypaint0_1.3.0-0dev2_amd64.deb
```

Then, when I tried to launch MyPaint 1.3.0, form a terminal, I got this in the terminal:

#### Backtraces or error messages
```
Mypaint version: 1.3.0-alpha+git.2c07a27b
System information: Linux-4.8.0-1-amd64-x86_64-with-debian-stretch-sid
Using: Python 2.7.13, GTK 3.22.4, GdkPixbuf 2.36.0, Cairo 1.14.8, GLib 2.50.0
Traceback (most recent call last):
  File "/home/digitalko/Tmp/mypaint/gui/application.py", line 381, _at_application_start(self=<gui.application.Application object>, filenames=[], fullscreen=None)
            col = self.brush_color_manager.get_color()
            self.brushmanager.select_initial_brush()
            self.brush_color_manager.set_color(col)
  variables: {'self.brushmanager.select_initial_brush': ('local', <bound method BrushManager.select_initial_brush of <gui.brushmanager.BrushManager object at 0x7f0145ddf250>>)}
  File "/home/digitalko/Tmp/mypaint/gui/brushmanager.py", line 430, select_initial_brush(self=<gui.brushmanager.BrushManager object>)
                initial_brush = self.get_default_brush()
            self.select_brush(initial_brush)
  variables: {'initial_brush': ('local', <ManagedBrush u'deevad/2B_pencil' p=None>), 'self.select_brush': ('local', <bound method BrushManager.select_brush of <gui.brushmanager.BrushManager object at 0x7f0145ddf250>>)}
  File "/home/digitalko/Tmp/mypaint/gui/brushmanager.py", line 737, select_brush(self=<gui.brushmanager.BrushManager object>, brush=<ManagedBrush u'deevad/2B_pencil' p=None>)
            # brush, amongst other things
            self.brush_selected(brush, brushinfo)
  variables: {'self.brush_selected': ('local', <MethodWithObservers gui.brushmanager.BrushManager.brush_selected>), 'brush': ('local', <ManagedBrush u'deevad/2B_pencil' p=None>), 'brushinfo': ('local', <lib.brush.BrushInfo object at 0x7f0145de2190>)}
  File "/home/digitalko/Tmp/mypaint/lib/observable.py", line 286, __call__(self=<MethodWithObservers gui.brushmanager.BrushManager.brush_selected>, *args=(<ManagedBrush u'deevad/2B_pencil' p=None>, <lib.brush.BrushInfo object>), **kwargs={})
                try:
                    observer(observed, *args, **kwargs)
                except BoundObserverMethod._ReferenceError:
  variables: {'observed': ('local', <gui.brushmanager.BrushManager object at 0x7f0145ddf250>), 'args': ('local', (<ManagedBrush u'deevad/2B_pencil' p=None>, <lib.brush.BrushInfo object at 0x7f0145de2190>)), 'observer': ('local', <BoundObserverMethod gui.brushmodifier.BrushModifier.brush_selected_cb>), 'kwargs': ('local', {})}
  File "/home/digitalko/Tmp/mypaint/lib/observable.py", line 469, __call__(self=<BoundObserverMethod gui.brushmodifier.BrushModifier.brush_selected_cb>, observed=<gui.brushmanager.BrushManager object>, *args=(<ManagedBrush u'deevad/2B_pencil' p=None>, <lib.brush.BrushInfo object>), **kwargs={})
                raise self._ReferenceError
            self._observer_func(observer, observed, *args, **kwargs)
            del observer
  variables: {'observed': ('local', <gui.brushmanager.BrushManager object at 0x7f0145ddf250>), 'args': ('local', (<ManagedBrush u'deevad/2B_pencil' p=None>, <lib.brush.BrushInfo object at 0x7f0145de2190>)), 'self._observer_func': ('local', <function brush_selected_cb at 0x7f0145e1ea28>), 'observer': ('local', <gui.brushmodifier.BrushModifier object at 0x7f013e535890>), 'kwargs': ('local', {})}
  File "/home/digitalko/Tmp/mypaint/gui/brushmodifier.py", line 225, brush_selected_cb(self=<gui.brushmodifier.BrushModifier object>, bm=<gui.brushmanager.BrushManager object>, managed_brush=<ManagedBrush u'deevad/2B_pencil' p=None>, brushinfo=<lib.brush.BrushInfo object>)
            b.end_atomic()
  variables: {'b.end_atomic': ('local', <bound method BrushInfo.end_atomic of <lib.brush.BrushInfo object at 0x7f0145e7de10>>)}
  File "/home/digitalko/Tmp/mypaint/lib/brush.py", line 439, end_atomic(self=<lib.brush.BrushInfo object>)
                for f in self.observers:
                    f(pending)
  variables: {'pending': ('local', set(['comment', 'tracking_noise', 'speed2_gamma', 'speed2_slowness', 'pressure_gain_log', 'stroke_holdtime', 'stroke_duration_logarithmic', 'slow_tracking', 'change_color_v', 'elliptical_dab_ratio', 'change_color_h', 'change_color_l', 'offset_by_speed_slowness', 'speed1_gamma', 'slow_tracking_per_dab', 'colorize', 'dabs_per_basic_radius', 'group', 'opaque', 'color_s', 'opaque_multiply', 'elliptical_dab_angle', 'lock_alpha', 'snap_to_pixel', 'eraser', 'opaque_linearize', 'stroke_threshold', 'description', 'color_h', 'parent_brush_name', 'smudge_radius_log', 'speed1_slowness', 'smudge', 'change_color_hsv_s', 'radius_by_random', 'dabs_per_second', 'color_v', 'custom_input', 'hardness', 'restore_color', 'smudge_length', 'radius_logarithmic', 'change_color_hsl_s', 'notes', 'custom_input_slowness', 'offset_by_speed', 'dabs_per_actual_radius', 'offset_by_random', 'direction_filter', 'anti_aliasing'])), 'f': ('local', <bound method CanvasRenderer.brush_modified_cb of <tileddrawwidget.CanvasRenderer object at 0x7f0145f722d0 (gui+tileddrawwidget+CanvasRenderer at 0x556e831c80f0)>>)}
  File "/home/digitalko/Tmp/mypaint/gui/tileddrawwidget.py", line 631, brush_modified_cb(self=<tileddrawwidget.CanvasRenderer object at 0x7f01...ileddrawwidget+CanvasRenderer at 0x556e831c80f0)>, settings=set(['anti_aliasing', 'change_color_h', 'change_color_hsl_s', 'change_color_hsv_s', 'change_color_l', 'change_color_v', ...]))
                # Reducing the number of updates is probably a good idea
                self.update_cursor()
  variables: {'self.update_cursor': ('local', <bound method CanvasRenderer.update_cursor of <tileddrawwidget.CanvasRenderer object at 0x7f0145f722d0 (gui+tileddrawwidget+CanvasRenderer at 0x556e831c80f0)>>)}
  File "/home/digitalko/Tmp/mypaint/gui/tileddrawwidget.py", line 593, update_cursor(self=<tileddrawwidget.CanvasRenderer object at 0x7f01...ileddrawwidget+CanvasRenderer at 0x556e831c80f0)>)
                radius, style = self._get_cursor_info()
                c = cursor.get_brush_cursor(radius, style, self.app.preferences)
            window.set_cursor(c)
  variables: {'self.app.preferences': ('local', {'frame.color_rgba': (0.12, 0.12, 0.12, 0.92), 'ui.feedback.scale': False, 'ui.hide_subwindows_in_fullscreen': True, 'ui.feedback.last_pos': False, 'ui.dark_theme_variant': True, 'brushmanager.selected_groups': [], 'ui.toolbar_items': {'toolbar1_file': True, 'toolbar1_view_modes': True, 'toolbar1_edit': True, 'toolbar1_blendmodes': False, 'toolbar1_view_manips': False, 'toolbar1_view_resets': True, 'toolbar1_scrap': False, 'toolbar1_linemodes': True}, 'saving.scrap_prefix': u'~/MyPaint/scrap', 'ui.hide_menubar_in_fullscreen': True, 'input.button_mapping': {'<Control>Button1': 'ColorPickMode', 'Button2': 'PanViewMode', '<Alt><Shift>Button2': 'FrameEditMode', '<Control>Button2': 'ZoomViewMode', '<Control><Shift>Button2': 'FrameEditMode', '<Shift>Button2': 'RotateViewMode', 'Button3': 'ShowPopupMenu', '<Shift>Button1': 'StraightMode', '<Alt>Button1': 'ColorPickMode', '<Alt>Button2': 'ZoomViewMode'}, 'ui.hide_toolbar_in_fullscreen': True, 'view.default_zoom': 1.0, 'misc.context_restores_color': True, 'input.global_pressure_mapping': [(0.0, 1.0), (1.0, 0.0)], 'workspace.layout': {'position': {'y': 32, 'h': -100, 'w': -50, 'x': 50}, 'autohide': True}, 'ui.parts': {'main_toolbar': True, 'menubar': True}, 'brushmanager.selected_brush': u'deevad/2B_pencil', 'ui.toolbar_icon_size': 'large', 'document.autosave_backups': True, 'display.colorspace': 'srgb', 'input.devices': {u'PenTablet  Pen:pen:2': {}, u'PenTablet  Pen (Eraser):eraser:2': {}, u'Wayland Pointer:mouse:2': {}}, 'view.real_alpha_checks': True, 'document.autosave_interval': 10, 'saving.default_format': 'openraster', 'scratchpad.last_opened_scratchpad': u'/tmp/mypaint_testconfig/scratchpads/autosave.ora', 'input.device_mode': 'screen'}), 'c': (None, []), 'style': ('local', 0), 'radius': ('local', 4.73400003322535), 'cursor.get_brush_cursor': ('global', <function get_brush_cursor at 0x7f01462b2320>)}
  File "/home/digitalko/Tmp/mypaint/gui/cursor.py", line 101, get_brush_cursor(radius=4.73400003322535, style=0, prefs={'brushmanager.selected_brush': u'deevad/2B_pencil', 'brushmanager.selected_groups': [], 'display.colorspace': 'srgb', 'document.autosave_backups': True, 'document.autosave_interval': 10, 'frame.color_rgba': (0.12, 0.12, 0.12, 0.92), 'input.button_mapping': {'<Alt><Shift>Button2': 'FrameEditMode', '<Alt>Button1': 'ColorPickMode', '<Alt>Button2': 'ZoomViewMode', '<Control><Shift>Button2': 'FrameEditMode', '<Control>Button1': 'ColorPickMode', '<Control>Button2': 'ZoomViewMode', '<Shift>Button1': 'StraightMode', '<Shift>Button2': 'RotateViewMode', 'Button2': 'PanViewMode', 'Button3': 'ShowPopupMenu'}, 'input.device_mode': 'screen', 'input.devices': {u'PenTablet  Pen (Eraser):eraser:2': {}, u'PenTablet  Pen:pen:2': {}, u'Wayland Pointer:mouse:2': {}}, 'input.global_pressure_mapping': [(0.0, 1.0), (1.0, 0.0)], ...})
            pixbuf = _image_surface_to_pixbuf(surf)
            last_cursor = Gdk.Cursor.new_from_pixbuf(display, pixbuf,
  variables: {'surf': ('local', <cairo.ImageSurface object at 0x7f013e4b8230>), '_image_surface_to_pixbuf': ('global', <function _image_surface_to_pixbuf at 0x7f01462b2398>), 'pixbuf': (None, [])}
  File "/home/digitalko/Tmp/mypaint/gui/cursor.py", line 111, _image_surface_to_pixbuf(surf=<cairo.ImageSurface object>)
        h = surf.get_height()
        return Gdk.pixbuf_get_from_surface(surf, 0, 0, w, h)
  variables: {'surf': ('local', <cairo.ImageSurface object at 0x7f013e4b8230>), 'h': ('local', 10), 'w': ('local', 10), 'Gdk.pixbuf_get_from_surface': ('global', gi.FunctionInfo(pixbuf_get_from_surface))}
KeyError: 'could not find foreign type Surface'
```
I did get a first error window message that I was able to close. But each time I tried to close the window, I got this second, looping, error message:

```
Mypaint version: 1.3.0-alpha+git.2c07a27b
System information: Linux-4.8.0-1-amd64-x86_64-with-debian-stretch-sid
Using: Python 2.7.13, GTK 3.22.4, GdkPixbuf 2.36.0, Cairo 1.14.8, GLib 2.50.0
Traceback (most recent call last):
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
```
#### Basic system details

MyPaint version: 3.0-alpha+git.2c07a27b.
Operating system: Debian SID
Desktop environment: GNOME 3.22 under Wayland

#### Steps to reproduce

1. Start MyPaint 3.0-alpha+git.2c07a27b.
2. Try to close the first error window message
3. Loop of the second error window message, when I try to close it, without being able to reach the "drawing area".

**BUT TODAY, I am no more able to reproduce this error.**

I do not know why :s

Maybe, today's packages upgrade solved the issue? Or something else? "(I tweaked libinput)

**Comment by achadwick** [comment]
@DigitalKO 

Version 3 is not even being planned yet. Please be more careful when you report version numbers; it's obvious what you mean in this in this instance, but if in doubt: copy and paste (Ctrl+C, Ctrl+V in Wayland), don't retype things. I assume you mean 1.3.0-alpha, because you included the 2c07a27b :smile: 

I cannot reproduce this error in Wayland on Debian unstable either. Do you have `python-gi-cairo` and `gir1.2-freedesktop` installed? It sounds like you are missing `cairo-1.0.typelib`, which is typically provided by the last one of those packages.

Right now, I'm not seeing a bug in MyPaint here.

**Comment by DigitalKO** [comment]
Sorry :s and yes I have these two packages installed, for long time ago:

```
ii  gir1.2-freedesktop:amd64                  1.50.0-1                  amd64                     Introspection data for some FreeDesktop components
ii  python-gi-cairo                           3.22.0-2                  amd64                     Python Cairo bindings for the GObject library
```

And for now, it works :)

I have only some warnings:

```
./mypaint
INFO: mypaint: Installation layout: not installed, testing from within the source tree
INFO: lib.i18n: POSIX: LANG='en_US.utf8'
INFO: lib.i18n: POSIX: LANGUAGE=None

** (mypaint:5235): WARNING **: Couldn't connect to accessibility bus: Failed to connect to socket /tmp/dbus-JzvpLY1s8t: Connection refused

(mypaint:5235): Gdk-WARNING **: /build/gtk+3.0-3.22.4/./gdk/wayland/gdkcursor-wayland.c:170: Unable to load arrow from the cursor theme

(mypaint:5235): Gdk-WARNING **: /build/gtk+3.0-3.22.4/./gdk/wayland/gdkcursor-wayland.c:170: Unable to load arrow from the cursor theme
/home/digitalko/Tmp/mypaint/gui/overlays.py:19: PyGIWarning: PangoCairo was imported without specifying a version first. Use gi.require_version('PangoCairo', '1.0') before import to ensure that the right version gets loaded.
  from gi.repository import PangoCairo
INFO: gui.device: New device 'Wayland Pointer' (GDK_SOURCE_MOUSE, axes:2, class=GdkWaylandDevice, vendor=None, product=None)
INFO: gui.device: New device 'PenTablet  Pen' (GDK_SOURCE_PEN, axes:2, class=GdkWaylandDevice, vendor=None, product=None)
INFO: gui.device: New device 'PenTablet  Pen (Eraser)' (GDK_SOURCE_ERASER, axes:2, class=GdkWaylandDevice, vendor=None, product=None)
INFO: gui.document: Initialized background from u'./backgrounds/10_soft_yellow.png'
WARNING: gui.keyboard: Ignoring keybinding for '<Actions>/BrushModifierActions/BlendModeMenu'

(mypaint:5235): Gtk-WARNING **: Theme parsing error: <data>:5:50: The style property GtkMenuBar:internal-padding is deprecated and shouldn't be used anymore. It will be removed in a future version

(mypaint:5235): Gtk-WARNING **: Allocating size to gui+workspace+_Paned 0x5564c1fc9250 without calling gtk_widget_get_preferred_width/height(). How does the code know the size to allocate?
INFO: lib.document: Using u'load_ora' to load u'/home/digitalko/.local/share/mypaint/scratchpads/autosave.ora' (kwargs={'feedback_cb': <function gtk_main_tick at 0x7f3cc1e57ed8>})
INFO: lib.document: load_ora: u'/home/digitalko/.local/share/mypaint/scratchpads/autosave.ora'
INFO: lib.document: 0.007s load_ora total
INFO: gui.filehandling: Loaded scratchpad from u'/home/digitalko/.local/share/mypaint/scratchpads/autosave.ora'

(mypaint:5235): Gtk-WARNING **: Allocating size to gui+workspace+_Paned 0x5564c1fc9250 without calling gtk_widget_get_preferred_width/height(). How does the code know the size to allocate?
```
